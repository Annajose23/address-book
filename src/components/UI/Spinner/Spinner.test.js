import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'; 

import Spinner from './Spinner';

configure({ adapter: new Adapter() });

describe('Spinner', ()=>{
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Spinner/>);
    });

    it('Should contain a div', () => {
        expect(wrapper.find('div').length).toEqual(1);
    });

    it('Should contain class name - loader', () => {
        expect(wrapper.find('div').hasClass('Loader')).toEqual(true);
    });

    it('render correctly - Spinner component', () => {  
        const SpinnerComponent = renderer.create(<Spinner/>).toJSON();
        expect(SpinnerComponent).toMatchSnapshot();
    });
})