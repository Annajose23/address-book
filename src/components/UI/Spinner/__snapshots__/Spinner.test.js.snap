// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner render correctly - Spinner component 1`] = `
<div
  className="Loader"
>
  Loading...
</div>
`;
